Auth0/CredentialsManager.swift

```diff
@@ -193,4 +192,42 @@ public struct CredentialsManager {
             }
         }
     }
+
+    private func hasExpired(_ credentials: Credentials) -> Bool {
+
+        if let expiresAT = credentials.expiresIn {
+            if expiresAT < Date() { return false }
+        }
+
+        if let idToken = credentials.idToken,
+            let jwt = decode(jwt: idToken),
+            let jwtExp = jwt["exp"] as? Double {
+                let expiresToken = Date(timeIntervalSince1970: jwtExp)
+                if expiresToken < Date() { return false }
+        }
+
+        return true
+    }
+}
+
+func decode(jwt: String) -> [String: Any]? {
```

Comment:
Is there no JWT decoding in this library currently? Would it make sense to pull in whatever lib might be used for the OIDC improvements?

Reply:
Just noticed this is from another class ... is this where it will be located later?